### PR TITLE
Added option to enable/disable deselect for date picker

### PR DESCRIPTION
**Pull request by mbulthuis**
I discused this feature with Tom Eugelink

### CODE REVIEW

**Comment by mbulthuis** [comment]
Note: only added supported for single select mode.

File: src/main/java/jfxtras/labs/scene/control/CalendarPicker.java [hunk 1]
```diff
@@ -53,16 +53,23 @@
  */
 public class CalendarPicker extends Control
 {
+    private boolean deselectEnabled = true;
```
Comment (tbee) [inline src/main/java/jfxtras/labs/scene/control/CalendarPicker.java]: In JavaFX we do not use Java Bean style properties, but JavaFX properties. This allows the binding functionality. See the show time property at line 209.

File: src/main/java/jfxtras/labs/internal/scene/control/skin/CalendarPickerControlSkin.java [hunk 2]
```diff
@@ -427,12 +427,6 @@ private void toggle(ToggleButton toggleButton, boolean shiftIsPressed)
 			{
 				lCalendars.remove(0);
 			}
-			// MULTIPLE: do nothing, just add the new one
-			//           if shift is pressed, behave like RANGE below
-			while (getSkinnable().getMode() == CalendarPicker.Mode.SINGLE && lCalendars.size() > 1)
```
Comment (tbee) [inline src/main/java/jfxtras/labs/internal/scene/control/skin/CalendarPickerControlSkin.java:432]: Why is this code removed? If I remember correctly it is needed to support the change of the Mode from multiple to one.

File: src/main/java/jfxtras/labs/scene/control/CalendarPicker.java [hunk 3]
```diff
@@ -53,16 +53,23 @@
...
 	/**
-	 * 
+	 * Default constructor
 	 */
 	public CalendarPicker()
 	{
 		construct();
 	}
+
+    public CalendarPicker(boolean enableDeSelect)
```
Comment (tbee) [inline src/main/java/jfxtras/labs/scene/control/CalendarPicker.java]: Not sure if we are going to create a separate constructor just for the disable deselect mode. It ain't such an important feature to warrant one. Better use new CalendarPicker().withEnableDeselect(false);

File: src/main/java/jfxtras/labs/internal/scene/control/skin/CalendarPickerControlSkin.java (hunk 2 above)
Comment (mbulthuis) [inline src/main/java/jfxtras/labs/internal/scene/control/skin/CalendarPickerControlSkin.java:432]: Look at the old version. This while loop is completely identical to the previous loop.

Comment (tbee) [inline src/main/java/jfxtras/labs/internal/scene/control/skin/CalendarPickerControlSkin.java:432]: I only see the removal of the loop in github.

**Pull request merged**